hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ConcurrentDistributionSetInvalidationTest.java

```diff
@@ -38,35 +48,54 @@
  */
 @Feature("Component Tests - Repository")
 @Story("Concurrent Distribution Set invalidation")
-@TestPropertySource(properties = "hawkbit.server.repository.dsInvalidationLockTimeout=1")
+@ContextConfiguration(classes = ConcurrentDistributionSetInvalidationTest.Config.class)
+@TestPropertySource(properties = { "hawkbit.server.repository.dsInvalidationLockTimeout=1" })
 public class ConcurrentDistributionSetInvalidationTest extends AbstractJpaIntegrationTest {
+    
+    @Configuration
+    static class Config {
+
+        /**
+         * Creates a {@link RolloutGroupRepository} bean that is slow during saving already created Groups.
+         * This gives this test more time to succeed
+         */
+        @Bean
+        @Primary
+        RolloutGroupRepository slowRolloutGroupRepository(final RolloutGroupRepository groupRepo) {
+            final RolloutGroupRepository slowGroupRepo = mock(RolloutGroupRepository.class, delegatesTo(groupRepo));
+
+            doAnswer(invocation -> {
+                final JpaRolloutGroup group = invocation.getArgument(0);
+                if (group.getId() == null) {
+                    return groupRepo.save(group);
+                }
+                TimeUnit.SECONDS.sleep(2);
```

Comment:
could you please elaborate/comment why 2 seconds are chosen here. Is there really no better approach as to wait before saving, doesn't it mask the real problem of quota calculation?..

Reply:
the idea here is to simulate a slow group repository, therefore I added the sleep construct 2 seconds are just a fitting number.
I am not sure I got your point about the quota calculation.
The test was successful previously because we create a big number of entities and groups, and so the ds invalidation could succeed, but since we reduced this number, the test is too fast for a concurrent invalidatoin